File: admin/src/api/erp_inventory_check_detail.ts
import { PaginatedRequest, PaginatedResponse } from '@/base/page';
import { api } from '@/utils/request';
import { ErpProductResponse } from './erp_product';

const apis = {
  create: '/erp/erp_inventory_check_detail/create', // 新增
  update: '/erp/erp_inventory_check_detail/update', // 修改
  delete: '/erp/erp_inventory_check_detail/delete', // 删除
  get: '/erp/erp_inventory_check_detail/get', // 单条查询
  list: '/erp/erp_inventory_check_detail/list', // 列表查询
  page: '/erp/erp_inventory_check_detail/page', // 分页查询
}

export interface ErpInventoryCheckDetailRequest {
  id?: number; // ID
  warehouse_id: number; // 仓库ID
  product_id: number; // 产品ID
  checked_quantity: number; // 盘点数量
  remarks?: string; // 备注

  product?: ErpProductResponse;
}

export interface ErpInventoryCheckDetailResponse {
  id: number; // ID
  order_id: number; // 盘点订单ID
  warehouse_id: number; // 仓库ID
  product_id: number; // 产品ID
  checked_quantity: number; // 盘点数量
  remarks: string; // 备注
  department_code: string; // 部门编码
  department_id: number; // 部门ID
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间
}

export interface ErpInventoryCheckDetailBaseResponse {
  id: number; // ID
  order_id: number; // 盘点订单ID
  warehouse_id: number; // 仓库ID
  product_id: number; // 产品ID
  checked_quantity: number; // 盘点数量
  remarks: string; // 备注

  product?: ErpProductResponse;
}

export interface ErpInventoryCheckDetailQueryCondition extends PaginatedRequest {

}

export const createErpInventoryCheckDetail = (erp_inventory_check_detail: ErpInventoryCheckDetailRequest): Promise<number> => {
  return api.post<number>(apis.create, erp_inventory_check_detail);
}

export const updateErpInventoryCheckDetail = (erp_inventory_check_detail: ErpInventoryCheckDetailRequest): Promise<void> => {
  return api.post<void>(apis.update, erp_inventory_check_detail);
}

export const deleteErpInventoryCheckDetail = (id: number): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
}

export const getErpInventoryCheckDetail = (id: number): Promise<ErpInventoryCheckDetailResponse> => {
  return api.get<ErpInventoryCheckDetailResponse>(`${apis.get}/${id}`);
}

export const listErpInventoryCheckDetail = (): Promise<Array<ErpInventoryCheckDetailResponse>> => {
  return api.get<Array<ErpInventoryCheckDetailResponse>>(apis.list);
}

export const pageErpInventoryCheckDetail = (condition: ErpInventoryCheckDetailQueryCondition): Promise<PaginatedResponse<ErpInventoryCheckDetailResponse>> => {
  return api.get<PaginatedResponse<ErpInventoryCheckDetailResponse>>(apis.page, condition);
}

File: server/module/module-erp/erp-server/src/convert/erp_inventory_check_detail.rs
use sea_orm::{Set, NotSet};
use crate::model::erp_inventory_check_detail::{self, Model as ErpInventoryCheckDetail, ActiveModel as ErpInventoryCheckDetailActiveModel};
use erp_model::request::erp_inventory_check_detail::{CreateErpInventoryCheckDetailRequest, UpdateErpInventoryCheckDetailRequest};
use erp_model::response::erp_inventory_check_detail::{ErpInventoryCheckDetailBaseResponse, ErpInventoryCheckDetailResponse};

pub fn create_request_to_model(request: &CreateErpInventoryCheckDetailRequest) -> ErpInventoryCheckDetailActiveModel {
    ErpInventoryCheckDetailActiveModel {
        warehouse_id: Set(request.warehouse_id.clone()),
        product_id: Set(request.product_id.clone()),
        checked_quantity: Set(request.checked_quantity.clone()),
        remarks: request.remarks.as_ref().map_or(NotSet, |remarks| Set(Some(remarks.clone()))),
        ..Default::default()
    }
}

pub fn update_add_request_to_model(request: &UpdateErpInventoryCheckDetailRequest) -> ErpInventoryCheckDetailActiveModel {
    ErpInventoryCheckDetailActiveModel {
        warehouse_id: Set(request.warehouse_id.clone()),
        product_id: Set(request.product_id.clone()),
        checked_quantity: Set(request.checked_quantity.clone()),
        remarks: request.remarks.as_ref().map_or(NotSet, |remarks| Set(Some(remarks.clone()))),
        ..Default::default()
    }
}

pub fn update_request_to_model(request: &UpdateErpInventoryCheckDetailRequest, existing: ErpInventoryCheckDetail) -> ErpInventoryCheckDetailActiveModel {
    let mut active_model: ErpInventoryCheckDetailActiveModel = existing.into();
    if let warehouse_id = &request.warehouse_id { 
        active_model.warehouse_id = Set(warehouse_id.clone());
    }
    if let product_id = &request.product_id { 
        active_model.product_id = Set(product_id.clone());
    }
    if let checked_quantity = &request.checked_quantity { 
        active_model.checked_quantity = Set(checked_quantity.clone());
    }
    if let Some(remarks) = &request.remarks { 
        active_model.remarks = Set(Some(remarks.clone()));
    }
    active_model
}

pub fn model_to_response(model: ErpInventoryCheckDetail) -> ErpInventoryCheckDetailResponse {
    ErpInventoryCheckDetailResponse { 
        id: model.id,
        order_id: model.order_id,
        warehouse_id: model.warehouse_id,
        product_id: model.product_id,
        checked_quantity: model.checked_quantity,
        remarks: model.remarks,
        department_code: model.department_code,
        department_id: model.department_id,
        creator: model.creator,
        create_time: model.create_time,
        updater: model.updater,
        update_time: model.update_time,
    }
}

pub fn model_to_base_response(model: ErpInventoryCheckDetail) -> ErpInventoryCheckDetailBaseResponse {
    ErpInventoryCheckDetailBaseResponse { 
        id: model.id,
        warehouse_id: model.warehouse_id,
        product_id: model.product_id,
        checked_quantity: model.checked_quantity,
        remarks: model.remarks,
    }
}